### ISSUE TITLE
incompatible with Julia 1.0

### ISSUE THREAD

**Issue by NLaws** [JuliaStochOpt/StochDynamicProgramming.jl#168]
Just tried to run one of the examples (awesome work! thank you for developing this package!) and the `import StochDynamicProgramming` gives:
```
[ Info: Precompiling StochDynamicProgramming [f6024387-75e5-5fdf-b095-327b05c0905f]
ERROR: LoadError: LoadError: UndefVarError: Nullable not defined
Stacktrace:
 [1] top-level scope at none:0
 [2] include at ./boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1044
 [4] include at ./sysimg.jl:29 [inlined]
 [5] include(::String) at /Users/nlaws/.julia/packages/StochDynamicProgramming/sNtGM/src/StochDynamicProgramming.jl:12
 [6] top-level scope at none:0
 [7] include at ./boot.jl:317 [inlined]
 [8] include_relative(::Module, ::String) at ./loading.jl:1044
 [9] include(::Module, ::String) at ./sysimg.jl:29
 [10] top-level scope at none:2
 [11] eval at ./boot.jl:319 [inlined]
 [12] eval(::Expr) at ./client.jl:393
 [13] top-level scope at ./none:3
in expression starting at /Users/nlaws/.julia/packages/StochDynamicProgramming/sNtGM/src/sdpLoops.jl:90
in expression starting at /Users/nlaws/.julia/packages/StochDynamicProgramming/sNtGM/src/StochDynamicProgramming.jl:13
ERROR: Failed to precompile StochDynamicProgramming [f6024387-75e5-5fdf-b095-327b05c0905f] to /Users/nlaws/.julia/compiled/v1.0/StochDynamicProgramming/hX8uh.ji.
Stacktrace:
 [1] macro expansion at ./logging.jl:311 [inlined]
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1190
 [3] macro expansion at ./logging.jl:309 [inlined]
 [4] _require(::Base.PkgId) at ./loading.jl:947
 [5] require(::Base.PkgId) at ./loading.jl:858
 [6] macro expansion at ./logging.jl:309 [inlined]
 [7] require(::Module, ::Symbol) at ./loading.jl:840
```

Here is my output from `versioninfo()`:
```
julia> versioninfo()
Julia Version 1.0.2
Commit d789231e99 (2018-11-08 20:11 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin14.5.0)
  CPU: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = atom -a
  JULIA_NUM_THREADS = 4
```

I don't know enough about Julia to know if this is related to https://github.com/JuliaStochOpt/StochDynamicProgramming.jl/issues/163, however my version of JuMP is less than 0.19:
```
julia> Pkg.installed()
Dict{String,Union{Nothing, VersionNumber}} with 6 entries:
  "Juno"                    => v"0.5.3"
  "Clp"                     => v"0.5.1"
  "Atom"                    => v"0.7.10"
  "Gurobi"                  => v"0.5.6"
  "StochDynamicProgramming" => v"0.5.0"
  "JuMP"                    => v"0.18.4"
```

**Comment by odow** [comment]
This package hasn't been updated to Julia v1 yet. If you need it, you should install Julia v0.6.

**Comment by NLaws** [comment]
@odow ok thanks for the response. Good to know the compatible Julia version so that I can start working with the package.

**Comment by odow** [comment]
Other options you could look into are https://github.com/JuliaStochOpt/StructDualDynProg.jl, which I think has been updated to Julia 1.0 (@blegat?); https://github.com/odow/SDDP.jl. which only works on Julia 0.6; and https://github.com/odow/Kokako.jl which only works on Julia 1.0 and is under development (and not documented well), but includes cool things like being able to solve infinite horizon problems.

**Comment by blegat** [comment]
> Other options you could look into are https://github.com/JuliaStochOpt/StructDualDynProg.jl, which I think has been updated to Julia 1.0 (@blegat?)

Yes :)

**Comment by NLaws** [comment]
Awesome. Thanks so much for the knowledge.

**Comment by frapac** [comment]
I am planning to update StochDynamicProgramming to julia 1.0+ and JuMP 0.19. Hope that we will able to release soon!

**Comment by trigaut** [comment]
> I am planning to update StochDynamicProgramming to julia 1.0+ and JuMP 0.19. Hope that we will able to release soon!

Need any help? :smiley:

**Comment by frapac** [comment]
Solved with #173
